### PR TITLE
Pipeline maintenance

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Changes the utils package directory from `airflow_utils` to `airqo_etl_utils`.
- [x] Added tenant to measurements sent to message broker.
- [x] Introduces an ETL pipeline for populating the data warehouse
- [x] Introduces an ETL pipeline for weather data.
- [x] Fixes a typo on the device/site utility functions
- [x] Addresses data issues of [MOB-7](https://airqoteam.atlassian.net/browse/MOB-7) 

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-1040](https://airqoteam.atlassian.net/browse/PLAT-1040)

**_HOW DO I TEST OUT THIS PR?_**
Follow instructions in the [README file](https://github.com/airqo-platform/AirQo-api/tree/PLAT-1040-stream-data-into-the-datawarehouse/src/airflow#running-using-docker)

**_ARE THERE ANY RELATED PRs?_**
#753

### CODE REVIEW

File: src/airflow/airqo_etl_utils/message_broker.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def send_data(
                 # partition = int(self.__partitions[partition_index])
                 range_data = data[i : i + 50]
 
-                message = {"data": range_data, "action": action}
+                message = {"data": range_data, "action": action, "tenant": tenant}
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/message_broker.py:41]: Added tenant cc @Baalmart

File: src/airflow/airqo_etl_utils/schema/data_warehouse.json [hunk 2]
```diff
@@ -0,0 +1,302 @@
+[
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/schema/data_warehouse.json:1]: Data warehouse schema

File: src/airflow/airqo_etl_utils/schema/weather_data.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
+[
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/schema/weather_data.json:1]: weather data schema

File: src/airflow/dags/app_insights.py [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 from airflow.decorators import dag, task
 
-from airflow_utils.commons import slack_dag_failure_notification
+from airqo_etl_utils.commons import slack_dag_failure_notification
 
 
 @dag(
     "App-Forecast-Insights",
-    schedule_interval=None,
+    schedule_interval="30 * * * *",
```
Comment (NoahNsimbe) [inline src/airflow/dags/app_insights.py:9]: scheduling ETL for app insights

File: src/airflow/dags/app_insights.py [hunk 5]
```diff
@@ -38,58 +37,144 @@ def load(data: dict):
 
 
 @dag(
-    "App-Daily-Insights",
-    schedule_interval="0 * * * *",
+    "App-Historical-Daily-Insights",
```
Comment (NoahNsimbe) [inline src/airflow/dags/app_insights.py:40]: DAG for the app historical insights data

File: src/airflow/dags/weather_measurements.py [hunk 6]
```diff
@@ -1,211 +1,87 @@
...
+        )
 
     extracted_data = extract()
-    transformed_data = transform(extracted_data)
-    load(transformed_data)
+    load(extracted_data)
 
 
 @dag(
     "Hourly-Weather-Measurements",
```
Comment (NoahNsimbe) [inline src/airflow/dags/weather_measurements.py:45]: Hourly weather data DAG

File: src/airflow/airqo_etl_utils/data_warehouse_utils.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
+import pandas as pd
+
+from airqo_etl_utils.airqo_api import AirQoApi
+from airqo_etl_utils.bigquery_api import BigQueryApi
+
+
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/data_warehouse_utils.py:6]: data warehouse ETL utility functions

File: src/airflow/airqo_etl_utils/data_warehouse_utils.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    biq_query_api = BigQueryApi()
+    columns = [
+        "time",
+        "site_id",
+        "device",
+        "device_number",
+        "latitude",
+        "longitude",
+        "pm2_5",
+        "s1_pm2_5",
```
Comment (Baalmart) [inline src/airflow/airqo_etl_utils/data_warehouse_utils.py:19]: love the usage of the s1_pm_25 going forward and differentiating it from the pm2_5. @NoahNsimbe

File: src/airflow/airqo_etl_utils/message_broker.py (hunk 1 above)
Comment (Baalmart) [reply src/airflow/airqo_etl_utils/message_broker.py:41]: Thanks, so the package or library you are using allows the usage of custom fields? @NoahNsimbe

File: src/airflow/airqo_etl_utils/schema/measurements.json [hunk 9]
```diff
@@ -0,0 +1,156 @@
+[
+    {
```
Comment (Baalmart) [inline src/airflow/airqo_etl_utils/schema/measurements.json:2]: What is the main difference between `measurements` and `datawarehouse` schemas? Could you provide some context?  @NoahNsimbe

File: src/data-mgt/python/devices-tranformation/transformation.py [hunk 10]
```diff
@@ -134,7 +134,7 @@ def map_sites_to_tahmo_station(self):
```
Comment (Baalmart) [inline src/data-mgt/python/devices-tranformation/transformation.py:134]: So this looks like a file changed from another microservice, intentional? @NoahNsimbe

**Review by Baalmart — COMMENTED** [review]
All looks good, just left a few comments for your review. @NoahNsimbe

File: src/data-mgt/python/devices-tranformation/transformation.py (hunk 10 above)
Comment (NoahNsimbe) [reply src/data-mgt/python/devices-tranformation/transformation.py:134]: yes @Baalmart. that is check box 5 in the PR description. `Fixes a typo on the device/site utility functions`

File: src/airflow/airqo_etl_utils/schema/measurements.json (hunk 9 above)
Comment (NoahNsimbe) [reply src/airflow/airqo_etl_utils/schema/measurements.json:2]: measurements are pm2_5, pm10 etc...datawarehouse has site details and weather data in addition to the measurements. You can compare the schemas below @Baalmart 

**Datawarehouse**
https://github.com/airqo-platform/AirQo-api/blob/PLAT-1040-stream-data-into-the-datawarehouse/src/airflow/airqo_etl_utils/schema/data_warehouse.json

**Measurements (Device Readings)**
https://github.com/airqo-platform/AirQo-api/blob/PLAT-1040-stream-data-into-the-datawarehouse/src/airflow/airqo_etl_utils/schema/measurements.json

**Weather data**
https://github.com/airqo-platform/AirQo-api/blob/PLAT-1040-stream-data-into-the-datawarehouse/src/airflow/airqo_etl_utils/schema/weather_data.json

File: src/airflow/airqo_etl_utils/message_broker.py (hunk 1 above)
Comment (NoahNsimbe) [reply src/airflow/airqo_etl_utils/message_broker.py:41]: I am not using a package or library to create the message. Everything is developed from scratch. I however use a package to connect to the message broker

**Pull request merged**